server/timestamp/timestamp.go

```diff
@@ -65,28 +69,40 @@ func GetOrCreateTimestamp(gun string, store storage.MetaStore, cryptoService sig
 		return nil, nil, err
 	}
 
-	_, snapshot, err := snapshot.GetOrCreateSnapshot(gun, store, cryptoService)
+	snapshotTime, snapshot, err := snapshot.GetOrCreateSnapshot(gun, store, cryptoService)
```

Comment:
Non-blocking:  since this is the only thing that calls `GetOrCreateSnapshot` now, would it make sense to pass the previous timestamp or snapshot checksum from the previous timestamp, so `GetOrCreateSnapshot` can just get the metadata by checksum rather than have to walk the timestamp again?

I don't think it affects functionality (since either `GetOrCreateSnapshot` would just get the same timestamp and walk to the right place, or it gets a newer timestamp, and this timestamp generation will fail when writing), but would save one extra database read and unmarshal.